front/app/modules/commercial/moderation/admin/containers/index.tsx

```diff
@@ -253,83 +236,104 @@ const Moderation = memo<Props & InjectedIntlProps>(({ className, intl }) => {
     },
   ]);
 
-  const handleOnSelectAll = useCallback(
-    (_event: React.ChangeEvent) => {
-      if (!processing) {
-        if (!isNilOrError(moderations)) {
-          setSelectedModerations(
-            selectedModerations.length < moderations.length ? moderations : []
-          );
-        }
+  const {
+    list: moderations,
+    pageSize,
+    moderationStatus,
+    currentPage,
+    lastPage,
+    onModerationStatusChange,
+    onPageNumberChange,
+    onPageSizeChange,
+    onModeratableTypesChange,
+    onProjectIdsChange,
+    onSearchTermChange,
+    onIsFlaggedChange,
+  } = useModerations({
+    pageSize: selectedPageSize,
+    moderationStatus: 'unread',
+  });
+  const moderationsWithActiveFlag = useModerations({
+    isFlagged: true,
+  });
+
+  const handleOnSelectAll = (_event: React.ChangeEvent) => {
```

Comment:
Removed all the useCallback instances in this file because I was debugging. I didn't add them back because I don't know if they add visible benefits (or just make the code more complicated). Then I did what I think is more correct: only do state setting in handle functions and then make requests via useModerations in useEffect. 

It turned out this was not causing the bug, but I think this is better code in general (data fetching shouldn't happen in handle functions).

Reply:
I agree! 👍